lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -122,12 +125,22 @@ def to_kubeclient_resource
 
     def validate_definition(kubectl, selector: nil)
```

Comment:
Nor for this PR: I think we'll be able to significantly shrink the size of this method via the `TaskConfigValidator` which is probably the right point to decide if this really belongs in this class or some new object.